### PR TITLE
Add argument to QamqpQueue

### PR DESCRIPTION
**Pull request by droidsyer**
Added a arguments table to queue, in order to set policy (i.e. x-limit-lenght).

### CODE REVIEW

**Comment by droidsyer** [comment]
Added a arguments table to queue, in order to set policy (i.e. x-limit-lenght).

File: src/qamqpqueue_p.h [hunk 1]
```diff
@@ -55,6 +55,8 @@ class QAmqpQueuePrivate: public QAmqpChannelPrivate,
     bool consumeRequested;
 
     Q_DECLARE_PUBLIC(QAmqpQueue)
+    private:
+    QAmqpTable arguments;
```
Comment (mbroadst) [inline src/qamqpqueue_p.h]: this is already the implementation for the dptr, no need to make this member private

File: src/qamqpqueue.cpp [hunk 2]
```diff
@@ -18,7 +18,8 @@ QAmqpQueuePrivate::QAmqpQueuePrivate(QAmqpQueue *q)
       declared(false),
       recievingMessage(false),
       consuming(false),
-      consumeRequested(false)
+      consumeRequested(false),
+      arguments(QAmqpTable())
```
Comment (mbroadst) [inline src/qamqpqueue.cpp]: no need to add a default constructor here

File: src/qamqpqueue.cpp [hunk 3]
```diff
@@ -328,10 +329,11 @@ int QAmqpQueue::options() const
     return d->options;
 }
 
-void QAmqpQueue::declare(int options)
+void QAmqpQueue::declare(int options, const QAmqpTable arguments)
```
Comment (mbroadst) [inline src/qamqpqueue.cpp]: this should be a const& passed in:
`void QAmqpQueue::declare(int options, const QAmqpTable &arguments)`

**Comment by mbroadst** [comment]
Other than the above comments, it looks good to me. It would be nice if you could add this for `QAmqpExchange` as well, as it seems to also support optional arguments on declare

**Pull request merged**